code/obj/item/gun/gun_parent.dm

```diff
@@ -45,7 +95,24 @@ var/list/forensic_IDs = new/list() //Global list of all guns, based on bioholder
 	var/add_residue = 0 // Does this gun add gunshot residue when fired (Convair880)?
 
 	var/charge_up = 0 //Does this gun have a charge up time and how long is it? 0 = normal instant shots.
+
+	/// Number of times to shoot the gun when asked to shoot
+	var/burst_count = 1
+	/// Time after clicking the gun before it'll allow you to click with the gun again
 	var/shoot_delay = 4
+	/// Time between shots in a burst
+	var/refire_delay = (0.7 DECI SECONDS)
+	/// If not 0, the bullet will shoot off course by between 0 and this number degrees
+	var/spread_angle = 0
+	/// List of firemodes, changes how the gun fires
+	/// structure: list(list("name" = "name of mode", "burst_count" = burst, "refire_delay" = refire_delay, "shoot_delay" = shoot_delay, "spread_angle" = spread_angle, "projectile" = null))
+	/// if "projectile" is blank, it'll use the projectile stored in the magazine
+	var/list/firemodes = list(list("name" = "single-shot", "burst_count" = 1, "refire_delay" = 0.7, "shoot_delay" = 4, "spread_angle" = 0, "projectile" = null))
```

Comment:
Before I review any further in this, I feel like this needs to be changed.
Either datumize firemodes, or at the very least, write helper macro(s), because having to work with a list of assoc lists like this is just really ugly when it doesn't need to be.